src/components/Toolbar/index.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react'
+import {
+  GridToolbarContainer,
+  GridToolbarExport,
+  GridToolbarColumnsButton,
+  GridToolbarFilterButton,
+  GridToolbarDensitySelector,
+  GridToolbarProps
+} from '@mui/x-data-grid'
+import { TextField, IconButton } from '@mui/material'
+import SearchIcon from '@mui/icons-material/Search'
+import ClearIcon from '@mui/icons-material/Clear'
+
+interface CustomToolbarProps extends GridToolbarProps {
+  searchTerm: string
+  onSearchChange: (value: string) => void
+  onSearch: () => void
+  onReset: () => void
+}
+
+const CustomToolbar: React.FC<CustomToolbarProps> = ({
+  searchTerm,
+  onSearchChange,
+  onSearch,
+  onReset
+}) => {
+  return (
+    <GridToolbarContainer>
+      <GridToolbarColumnsButton />
+      <GridToolbarFilterButton />
+      <GridToolbarDensitySelector />
+      <GridToolbarExport />
+
+      <TextField
```

Comment:
i would align all this elements related to search to the headers right , it should work with some `align-items: flex-end`  i suppose